Backend/Tournament_User/TournamentUserMVSC/controllers/tournamentUserController.js

```diff
@@ -0,0 +1,366 @@
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+
+/**
+ * Get UserTournamentGameResult 
+ * @async
+ * @param {Object} req.params - The request parameters
+ * @param {string} req.params.UUID - The ID of the User.
+ * @param {string} req.params.tournamentId - The ID of the tournament.
+ * @returns {Json} Return incoming tournament based on upcoming dates.
+ * @throws {error} If input invalid or missing fields.
+ */
+
+exports.getUserTournamentGameRank = async (req, res, next) => {
+    const { UUID } = req.body;
+    const { tournamentId } = req.params;
+
+    try {
+        // Step 1: Check if tournament exists
+        const tournamentExists = await TournamentService.checkTournamentExists(tournamentId);
+        
+        if (!tournamentExists) {
+            const error = new Error('Tournament not found.');
+            error.statusCode = 404;  // Not Found
+            return next(error);  // Pass the error to middleware
+        }
+
+        // Step 2: Get user's tournament game rank if tournament exists
+        const getUserTournamentGameRank = await TournamentService.getUserTournamentGameRank(tournamentId, UUID);
+        
+        if (!getUserTournamentGameRank || getUserTournamentGameRank.length === 0) {
+            const error = new Error('No Game result found for the provided tournamentID.');
+            error.statusCode = 404;  // Not Found
+            return next(error);
+        }
+
+        // Respond with the ranking information
+        res.status(200).json({
+            
+            message: 'Current Tournament Ranking:',
+            success: true,
+            content: getUserTournamentGameRank
+        });
+    } catch (err) {
+        console.error('Error retrieving current ranking in the tournament:', err);
+        next(err);  // Pass the error to the middleware
+    }
+};
+
+
+/**
+ * Get the list of UUID players in the specified tournament.
+ * @async
+ * @param {Object} req.params - The request parameters
+ * @param {string} req.params.tournamentId - The ID of the tournament.
+ * @returns {Json} Returns a list of UUIDs participating in the tournament.
+ * @throws {error} If tournamentID invalid.
+ */
+
+
+// Get upcoming tournaments
+exports.getPlayersInTournament = async (req, res, next) => {
+    const { tournamentId } = req.params;
+
+    if (!tournamentId) {
+        const error = new Error('Missing required fields tournamentID');
+        error.statusCode = 400;
+        return next(error);
+    }
+    try {
+        //Check if tournament exist else return relavant error message.
+        const tournamentExists = await TournamentService.checkTournamentExists(tournamentId);
+        if (!tournamentExists) {
+            const error = new Error('Invalid tournamentID: Tournament not found.');
+            error.statusCode = 404;
+            return next(error);
+        }
+
+
+        const GetPlayersInTournament = await TournamentService.getPlayersInTournament(tournamentId);
```

Comment:
no need for constants here imo can use the `let result = ...` since the result isn't reused anywhere else in the code, got multiple of these instances if u can go thru dem n fix